app/assets/javascripts/discourse/app/models/group.js

```diff
@@ -463,13 +463,16 @@ export default class Group extends RestModel {
       data.category_id = parseInt(opts.categoryId, 10);
     }
 
-    return ajax(`/groups/${this.name}/${type}.json`, { data }).then((posts) => {
-      return posts.map((p) => {
-        p.user = User.create(p.user);
-        p.topic = Topic.create(p.topic);
-        p.category = Category.findById(p.category_id);
-        return EmberObject.create(p);
-      });
+    const posts = await ajax(`/groups/${this.name}/${type}.json`, { data });
+    const categories = await Category.asyncFindByIds(
+      posts.map((p) => p.category_id)
+    );
```

Comment:
I'll like to explore some of the thinking behind this way of loading categories information because we could just sideload categories information in the response of the first request.